pkg/server/server.go

```diff
@@ -508,14 +509,14 @@ func getAgentIdentifiers(stream agent.AgentService_ConnectServer) (pkgagent.Iden
 	return agentIdentifiers, nil
 }
 
-func (s *ProxyServer) validateAuthToken(token string) error {
+func (s *ProxyServer) validateAuthToken(ctx context.Context, token string) error {
 	trReq := &authv1.TokenReview{
 		Spec: authv1.TokenReviewSpec{
 			Token:     token,
 			Audiences: []string{s.AgentAuthenticationOptions.AuthenticationAudience},
 		},
 	}
-	r, err := s.AgentAuthenticationOptions.KubernetesClient.AuthenticationV1().TokenReviews().Create(trReq)
+	r, err := s.AgentAuthenticationOptions.KubernetesClient.AuthenticationV1().TokenReviews().Create(ctx, trReq, metav1.CreateOptions{})
```

Comment:
This is the only use of client-go in the code base. I think this client call works for older apiservers.